torchtune/models/llama4/_model_builders.py

```diff
@@ -0,0 +1,177 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+from typing import Optional
+
+from torchtune.data._prompt_templates import _TemplateType
+
+from torchtune.models.llama4 import llama4_decoder, llama4_vision_encoder
+from torchtune.models.llama4._tokenizer import LLAMA4_SPECIAL_TOKENS
+from torchtune.models.llama4._transform import Llama4Transform
+
+from torchtune.modules.model_fusion import EarlyFusionModel
+
+"""
+Model builders build specific instantiations using component builders. For example
+the llama4 model builder uses the MoE component builder to create the
+Llama4 MoE model.
+"""
+
+
+def llama4_scout_17b_16e(
+    decoder_trainable: bool = True,
+    vision_encoder_trainable: bool = False,
+    fusion_trainable: bool = True,
+    image_size: int = 336,
+) -> EarlyFusionModel:
+    """
+    Builder for creating an instance of the Llama4 Scout 17Bx16E model
+
```

Comment:
```suggestion
    Args:
        decoder_trainable (bool): Whether to make decoder params trainable. Default is True.
        encoder_trainable (bool): Whether to make encoder params trainable. Default is False.
        fusion_trainable (bool): Whether to make fusion params trainable. Default is True.
        image_size (int): Base image size that images will be tiled and resized to.
            Default is 336.

```
give me `Args` or give me death